### PR TITLE
Magnite5/phoenixsupport

### PR DESCRIPTION
**Pull request by magnite5**
### 🛠 Proposed Changes:

[Briefly Describe the changes introduced by this pull request]

### 📝 Details:

- [Detail 1]

### 🔗 Related Issue(s):

- Resolves #[Issue Number]

### CODE REVIEW

File: constants/RobotConstants.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+class RobotConstants:
+  # !
+  maxSpeed = 1
+  scale_speed = 1
+  rotationScale = 0.5
+  rotationCurvature = 2.0
+  rotationDeadzone = 0.075
+
+  isSimulation = False
```
Comment (NathanGrenier) [inline constants/RobotConstants.py:12]: There's a function to determine this. `wpilib.RobotBase.isReal()`

File: swerve/SwerveModule.py [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+
+    self.resetEncoders()
+
+    self.initNetworkTables()
+
+    self.lastDesiresState = self.getState()
+
+  def initNetworkTables(self):
+    network_table_instance = NetworkTableInstance.getDefault()
+    self.swerve_dashboard = network_table_instance.getTable(
```
Comment (NathanGrenier) [inline swerve/SwerveModule.py:115]: Not sure if we'd do this over the network table.

If you use the shuffleboard, the data will be published to the network table and visually to the shuffleboard.

File: swerve/SwerveSubsystems.py [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+      self.front_left.getPosition(),
+      self.front_right.getPosition(),
+      self.back_left.getPosition(),
+      self.back_right.getPosition(),
+    )
+
+  def periodic(self) -> None:
+    # TODO print gyro angle, robot pose on dashboard
+
+    if not RobotConstants.isSimulation:
```
Comment (NathanGrenier) [inline swerve/SwerveSubsystems.py:150]: Replace this with the proper way of determining if the code is running in a simulation (described above in another comment).

File: swerve/SwerveSubsystems.py [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+      self.back_right.getPosition(),
+    )
+
+  def stop(self) -> None:
+    self.front_left.stop()
+    self.front_right.stop()
+    self.back_left.stop()
+    self.back_right.stop()
+
+    if RobotConstants.isSimulation:
```
Comment (NathanGrenier) [inline swerve/SwerveSubsystems.py:167]: Replace this with the proper way of determining if the code is running in a simulation (described above in another comment).

**Review by NathanGrenier — CHANGES_REQUESTED** [review]

File: .pre-commit-config.yaml
Comment (NathanGrenier) [inline .pre-commit-config.yaml:1]: Keep this. It's just a config file to install the pre-commit hook. 

If you don't want any pre-commit functionality, don't install it via the `pre-commit` command. If it's already installed, delete the hook from the hidden `.git` folder.

**Comment by NathanGrenier** [comment]
Shouldn't this branch be merged into the `develop` branch?

**Review by Portnord — CHANGES_REQUESTED** [review]
Is this a mentor branch? Who's Magnite5?

**Comment by NathanGrenier** [comment]
Is this a mentor branch? Who's Magnite5?

**Comment by Portnord** [comment]
Closing this request. Work has been transferred to and continues on feature/drivebase.

**Pull request closed without merging**